docs/modules/ROOT/pages/api/erc1155.adoc

```diff
@@ -0,0 +1,664 @@
+
+TIP: For an overview of ERC1155, read our xref:erc1155.adoc[ERC1155 guide].
+
+== Core
+
+[.contract]
+[[IERC1155]]
+=== `++IERC1155++` link:https://github.com/OpenZeppelin/cairo-contracts/blob/cairo-2/src/token/erc1155/interface.cairo[{github-icon},role=heading-link]
+
+[.hljs-theme-dark]
+```javascript
+use openzeppelin::token::erc1155::interface::IERC1155;
+```
+Interface of the IERC1155 standard as defined in {eip1155}.
+
+[.contract-index]
+.{inner-src5}
+--
+0x6114a8f75559e1b39fcba08ce02961a1aa082d9256a158dd3e64964e4b1b52
+--
+
+[.contract-index]
+.Functions
+--
+* xref:#IERC1155-balance_of[`++balance_of(account, token_id)++`]
+* xref:#IERC1155-balance_of_batch[`++balance_of_batch(accounts, token_ids)++`]
+* xref:#IERC1155-safe_transfer_from[`++safe_transfer_from(from, to, token_id, value, data)++`]
+* xref:#IERC1155-safe_batch_transfer_from[`++safe_batch_transfer_from(from, to, token_ids, values, data)++`]
+* xref:#IERC1155-is_approved_for_all[`++is_approved_for_all(owner, operator)++`]
+* xref:#IERC1155-set_approval_for_all[`++set_approval_for_all(operator, approved)++`]
+--
+
+[.contract-index]
+.Events
+--
+* xref:#IERC1155-TransferSingle[`++TransferSingle(operator, from, to, id, value)++`]
+* xref:#IERC1155-TransferBatch[`++TransferBatch(operator, from, to, ids, values)++`]
+* xref:#IERC1155-ApprovalForAll[`++ApprovalForAll(owner, operator, approved)++`]
+* xref:#IERC1155-URI[`++URI(value, id)++`]
+--
+
+==== Functions
+
+[.contract-item]
+[[IERC1155-balance_of]]
+==== `[.contract-item-name]#++balance_of++#++(account: ContractAddress, token_id: u256) → u256++` [.item-kind]#external#
+
+Returns the amount of `token_id` owned by `account`.
+
+[.contract-item]
+[[IERC1155-balance_of_batch]]
+==== `[.contract-item-name]#++balance_of_batch++#++(accounts: Span<ContractAddress>, token_ids: Span<u256>) → Span<u256>++` [.item-kind]#external#
+
+Returns a list of balances derived from the `accounts` and `token_ids` pairs.
+
+[.contract-item]
+[[IERC1155-safe_transfer_from]]
+==== `[.contract-item-name]#++safe_transfer_from++#++(from: ContractAddress, to: ContractAddress, token_id: u256,  value: u256, data: Span<felt252>)++` [.item-kind]#external#
+
+Transfers `value` of `token_id` from `from` to `to`, checking first that `to` is aware of the ERC1155 protocol to prevent tokens being locked forever.
+For information regarding how contracts communicate their awareness of the ERC1155 protocol, see {receiving-tokens}.
+
+Emits a <<IERC1155-TransferSingle,TransferSingle>> event.
+
+[.contract-item]
+[[IERC1155-safe_batch_transfer_from]]
+==== `[.contract-item-name]#++safe_batch_transfer_from++#++(from: ContractAddress, to: ContractAddress, token_ids: Span<u256>, values: Span<u256>, data: Span<felt252>)++` [.item-kind]#external#
+
+Transfer `values` of `token_ids` (matched by index) from `from` to `to`, checking first that `to` is aware of the ERC1155 protocol to prevent tokens being locked forever.
```

Comment:
same here regarding awareness

Reply:
maybe we can talk about compatibility

Reply:
another alternative for this specific method is to say it's the batch version of `safe_transfer_from`

Reply:
> maybe we can talk about compatibility

Not sure what you mean here.

> another alternative for this specific method is to say it's the batch version of safe_transfer_from

I like explaining the behavior more explicitly. Sentences like "batched version of" could be ambiguous sometimes. like

Reply:
> Not sure what you mean here.

i mean that there's not really "awareness" of ERC1155 in the case of accounts, but "ERC1155 compatibility" instead. awareness is what `ERC1155Receiver`s have but that's only part of the story

Reply:
> I like explaining the behavior more explicitly. Sentences like "batched version of" could be ambiguous sometimes.

sure, i usually vouch for explicitness in API docs because it's a reference material: you don't read all but jump to what you're interested instead. i was only suggesting it because that's how we define it in the in-code docs and [the solidity one](https://docs.openzeppelin.com/contracts/5.x/api/token/erc1155#IERC1155-balanceOfBatch-address---uint256---)

Reply:
Updated already, let me know what you think.

Reply:
Removed the awareness as suggested.

Reply:
oh i thought you were referring to the batched definition